storage/remote/graphite/client.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package graphite
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"net"
+	"net/url"
+	"regexp"
+	"sort"
+	"time"
+
+	"github.com/prometheus/common/log"
+
+	"github.com/prometheus/common/model"
+)
+
+var (
+	// https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/grammar.py#L83
+	// Accept basically letters, numbers, and some escaped symbols.
+	symbolsRe      = regexp.MustCompile(`([\(\)\{\},=.'"\\])`)
+	illegalCharsRE = regexp.MustCompile(`[^a-zA-Z0-9\(\)\{\},=.'"\\]`)
```

Comment:
Are these all valid characters in both Windows and Unix filenames? https://github.com/prometheus/client_python/blob/master/prometheus_client/bridge/graphite.py#L14 is what I use over in the Python client.